halyard-web/src/main/java/com/netflix/spinnaker/halyard/Main.java

```diff
@@ -31,7 +31,8 @@
 @EnableAutoConfiguration
 @EnableConfigServer
 public class Main extends SpringBootServletInitializer {
-  private static final Map<String, Object> DEFAULT_PROPS = new DefaultPropertiesBuilder().build();
+  private static final Map<String, Object> DEFAULT_PROPS =
+      new DefaultPropertiesBuilder().property("spring.config.on-not-found", "ignore").build();
```

Comment:
I suspect that [DefaultPropertiesBuilder](https://github.com/spinnaker/kork/blob/ab04aa91ae6f7ab2dacb41c98db56bfa6fb0123e/kork-config/src/main/java/com/netflix/spinnaker/kork/boot/DefaultPropertiesBuilder.java) in kork is a more correct place for this, but I'm not confident about what's correct for the other spinnaker microservices.  Given https://github.com/spinnaker/spinnaker/issues/6745, I'd like to go ahead with this until we figure out something better.

Reply:
Agree, I wonder what the flow on effects are to loading other files.

> If you want to treat all locations as optional you can set spring.config.on-not-found=ignore in SpringApplication.setDefaultProperties(…​) or with a system/environment variable.

And `/opt/spinnaker/config/` is defined in [ConfigServerBootstrap.java](https://github.com/spinnaker/kork/blob/d1eb57a94d9a585a338e5c0eada04b93c5603b96/kork-config/src/main/java/com/netflix/spinnaker/kork/configserver/ConfigServerBootstrap.java#L26).

Reply:
I suspect we're getting away without this in clouddriver, because [`@EnableConfigServer`](https://github.com/spinnaker/clouddriver/blob/53c2eeda190d2195354851adfef5b93caf06b658/clouddriver-configserver/src/main/java/com/netflix/spinnaker/clouddriver/cache/CloudConfigRefreshConfig.java#L45) happens conditionally, and only when [RemoteConfigSourceConfigured](https://github.com/spinnaker/kork/blob/ab04aa91ae6f7ab2dacb41c98db56bfa6fb0123e/kork-cloud-config-server/src/main/java/com/netflix/spinnaker/kork/configserver/autoconfig/RemoteConfigSourceConfigured.java) returns true, which I suspect it doesn't by default...and so when people do enough configuration to enable the config server, they also set e.g. `spring.profiles.active = native` or whatever it takes to get clouddriver to start.